src/utils/embedding_utils.py

```
@@ -99,11 +102,11 @@ async def _send_batch_request(client: httpx.AsyncClient, batch_texts: List[str])
                 
         except httpx.RequestError as e:
             if attempt < max_retries - 1:
-                print(f"Error in batch request (attempt {attempt + 1}/{max_retries}): {e}")
+                logger.warning(f"Error in batch request (attempt {attempt + 1}/{max_retries}): {e}")
```

Ruff (0.12.2)

Logging statement uses f-string

(G004)